modules/reitit-ring/src/reitit/ring/coercion.cljc

```diff
@@ -24,24 +24,33 @@
   and :parameters from route data, otherwise does not mount."
   {:name ::coerce-request
    :spec ::rs/parameters
-   :compile (fn [{:keys [coercion parameters]} opts]
+   :compile (fn [{:keys [coercion parameters] :as route-data} opts]
               (cond
                 ;; no coercion, skip
                 (not coercion) nil
                 ;; just coercion, don't mount
                 (not parameters) {}
                 ;; mount
                 :else
-                (if-let [coercers (coercion/request-coercers coercion parameters opts)]
-                  (fn [handler]
-                    (fn
-                      ([request]
-                       (let [coerced (coercion/coerce-request coercers request)]
-                         (handler (impl/fast-assoc request :parameters coerced))))
-                      ([request respond raise]
-                       (let [coerced (coercion/coerce-request coercers request)]
-                         (handler (impl/fast-assoc request :parameters coerced) respond raise)))))
-                  {})))})
+                (try
+                  (if-let [coercers (coercion/request-coercers coercion parameters opts)]
+                    (fn [handler]
+                      (fn
+                        ([request]
+                         (let [coerced (coercion/coerce-request coercers request)]
+                           (handler (impl/fast-assoc request :parameters coerced))))
+                        ([request respond raise]
+                         (let [coerced (coercion/coerce-request coercers request)]
+                           (handler (impl/fast-assoc request :parameters coerced) respond raise)))))
+                    {})
+                  (catch Exception e
+                    (throw (ex-info (str "Coerce-request-middleware failed: " (.getMessage e))
```

Comment:
`reitit/exception/fail!` could be used instead?

Reply:
It doesn't currently take cause as parameter.